ibm/mas_devops/roles/db2/tasks/upgrade/main.yml

```diff
@@ -0,0 +1,193 @@
+---
+# 1. Look up the default channel for the db2u-operator package manifest
+# -----------------------------------------------------------------------------
+- name: Lookup db2u-operator packagemanifest
+  when: db2_channel is not defined or db2_channel == ""
+  kubernetes.core.k8s_info:
+    api_version: packages.operators.coreos.com/v1
+    kind: PackageManifest
+    name: db2u-operator
+    namespace: ibm-common-services
+  register: db2u_manifest_info
+
+- name: Set db2u-operator update channel to latest default channel if not provided
+  when: db2_channel is not defined or db2_channel == ""
+  set_fact:
+    db2_channel: "{{ db2u_manifest_info.resources[0].status.defaultChannel }}"
+
+- name: Set db2 version when not provided
+  when: db2_version is not defined or db2_version == ""
+  ansible.builtin.set_fact:
+    db2_version: "{{ db2u_manifest_info.resources[0].status.channels | selectattr('name', 'equalto', db2_channel) | map(attribute='currentCSVDesc') | map(attribute='annotations') | map(attribute='productVersion') | first }}"
+
+
+# Fail if db2_channel and/or version not discovered
+# -----------------------------------------------------------------------------
+- name: "Verify db2_channel and db2_version set"
+  assert:
+    that:
+      - db2_channel is defined and db2_channel != ""
+      - db2_version is defined and db2_version != ""
+    fail_msg: "Unable to determine db2_channel and/or db2_version from catalog"
+
+
+- name: Debug Db2 Universal Operator Install
+  ansible.builtin.debug:
+    msg:
+      - "Db2 Channel ............................ {{ db2_channel }}"
+      - "Db2 Version ............................ {{ db2_version }}"
+
+
+# 2. Lookup current db2u channel
+# -----------------------------------------------------------------------------
+- name: "Get the current subscription channel for db2"
+  kubernetes.core.k8s_info:
+    api_version: operators.coreos.com/v1alpha1
+    name: ibm-db2u-operator
+    namespace: ibm-common-services
+    kind: Subscription
+  register: db2_sub_info
+
+- name: "Set the value for the current channel for db2"
+  set_fact:
+    old_db2_channel: "{{ db2_sub_info.resources[0].spec.channel }}"
+
+- name: Debug Db2 Operator Update
+  ansible.builtin.debug:
+    msg:
+      - "Current Db2 Channel .................... {{ old_db2_channel }}"
+      - "Update Db2 Channel ..................... {{ db2_channel }}"
+      - "Update Db2 Engine Version .............. {{ db2_version }}"
+
+
+# 3. If the current operator channel is the same for the upgrade, no upgrade
+# -----------------------------------------------------------------------------
+- name: "Debug when no upgrade is needed"
+  when: db2_channel == old_db2_channel
+  debug:
+    msg: "No action is required. Db2 subscription is already on the {{ db2_channel }} channel"
+
+
+# 4. Create operand request
+# -----------------------------------------------------------------------------
+- name: "Create Db2 Operand Request"
```

Comment:
Can we fix the name of this "Delete old OperandRequest" .. it's very confusing that it still says "Create", when it's actually now just a cleanup step.

Reply:
Changed